api/src/main/java/org/xwiki/apps/googleapps/CookieAuthenticationPersistenceStoreTools.java

```diff
@@ -19,7 +19,261 @@
  */
 package org.xwiki.apps.googleapps;
 
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.SecretKeySpec;
+import javax.servlet.http.Cookie;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.web.XWikiRequest;
+
+/** Copied code from xwiki-authenticator-trusted
+ * https://github.com/xwiki-contrib/xwiki-authenticator-trusted/edit/master\
+ *   /xwiki-authenticator-trusted-api/src/main/java/org/xwiki/contrib/authentication\
+ *   /internal/CookieAuthenticationPersistenceStore.java.
+ * @version $Id$
+ * @since 2.5-RC1
```

Comment:
We can use directly a final version, `2.5` as the  `RC` versions would be released only for testing purposes.

Reply:
Did we not say 3.0 ?

Reply:
We did.